[exporter/datasetexporter] Upgrade to dataset-go v0.19.0

martin-majlis-s1 opened the pull request:
**Description:** This PR is upgrading dataset-go to version v0.19.0 - https://github.com/scalyr/dataset-go/releases/tag/v0.19.0

This PR is also fixing:
* #33498 - config validation is not called during conversion, therefore I had to remove tests from logs and trace exporter. These tests are now moved to the factory.
* #32533 - fixing failing integration tests, that were never updated and therefore the configuration contained invalid values

**Link to tracking Issue:** #33498, #32533, #33675

**Testing:** Unit tests and integration tests.

**Documentation:** <Describe the documentation added.>

fixes #32533

martin-majlis-s1 committed:
[exporter/datasetexporter] Upgrade to dataset-go v0.19.0

martin-majlis-s1 committed:
Add pointer to the PR

martin-majlis-s1 committed:
Do not generate life-cycle tests

crobert-1 commented on cmd/otelcontribcol/go.mod:
```diff
@@ -4,7 +4,7 @@ module github.com/open-telemetry/opentelemetry-collector-contrib/cmd/otelcontrib
 go 1.21.0
 
-toolchain go1.21.11
+toolchain go1.22.4
```
```suggestion
toolchain go1.21.11
```
This is causing the GitHub action `make genotelcontribcol` failure. Your local environment is running the command on `go1.22.4`, while GitHub is running against `go1.21.11`. When the GitHub action generates this file it generates it with the older toolchain version. When you run it locally, it generates it with your local golang version.

martin-majlis-s1 committed:
Use expected version of go.

Co-authored-by: Curtis Robert <crobert@splunk.com>

martin-majlis-s1 committed:
Merge branch 'main' into datasetexporter-update-to-0.19.0

martin-majlis-s1 committed:
Merge branch 'main' into datasetexporter-update-to-0.19.0

martin-majlis-s1 committed:
Merge branch 'main' into datasetexporter-update-to-0.19.0

tomaz-s1 commented on .chloggen/datasetexporter-update-to-0.19.0.yaml:
```diff
@@ -0,0 +1,27 @@
+component: datasetexporter
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Upgrade dataset-go to v0.19.0 and fix found issues
```
May want to briefly clarity what those "found issues" were under `subtext`.

tomaz-s1 reviewed (approved):
Small comment on expanding the changelog entry, besides that, LGTM.

Thanks.

martin-majlis-s1 committed:
Add more details to the changelog

martin-majlis-s1 commented:
> Small comment on expanding the changelog entry, besides that, LGTM.
> 
> Thanks.

I have added those details. My expectation was, that changelog is meant for users - and they do not care whether integration tests are now running when in the past they have been skipped.

zdaratom-s1 reviewed (approved):
LGTM

martin-majlis-s1 committed:
Merge branch 'main' into datasetexporter-update-to-0.19.0

crobert-1 committed:
[chore] Remove unused internal code (#33690)

This file appears to be no longer used. Since it's in an `internal`
directory it's not exposed publicly, so I don't believe it needs a
changelog.

For context, I believe its original use was to be consumed by
`cmd/configschema`, which has since been
[deleted](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/33384).
I believe we simply missed this file in the removal process.

dmitryax committed:
[chore] [receiver/mongodb] Remove redundant resource grouping (#33691)

The produced metrics are unnecessarily grouped by empty resources. This
change removes the excessive grouping to keep only unique resources

kernelpanic77 committed:
[pkg/ottl] Add schema_url field to contexts (#31444)

This PR attempts to make schemaURL for TransformContexts. 

**Description:** <Describe what has changed.>
Added a breaking change to creation of TransformContext function for
logs. Also made changes to all references of the function.

**Link to tracking Issue:** <Issue number if applicable>
#30229

---------

Co-authored-by: Evan Bradley <11745660+evan-bradley@users.noreply.github.com>
Co-authored-by: Tyler Helmuth <12352919+TylerHelmuth@users.noreply.github.com>

codeboten committed:
[chore] fix testbed runner (#33711)

This has been broken for some time as the definition of GOTEST was
removed

Signed-off-by: Alex Boten <223565+codeboten@users.noreply.github.com>

crobert-1 committed:
[chore][receiver/redis] Update username/password README description (#33710)

A recent issued brought up that the `username` config option is
available, but not documented in the README. The `password` option is
described in the README, but doesn't include information relevant for
Redis >= 6.0. This adds information on both config options.

Fixes https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/33707

pjanotti committed:
Add @pjanotti as codeowner for Windows perf counter and related components (#33695)

Adding myself as a codeowner for Windows performance counter and related
components per respective call for more volunteers on the respective
README files.

I have experience with Windows and was a maintainer of the collector
some years ago.

Per DM I am also removing @djaglowski from `receiver\windowseventlog`,
ActiveDirectoryDS, and IIS.

cc @alxbl

jmacd committed:
Add OTel-Arrow components to contribcol; mark these alpha (#33628)

**Description:** Labels the `otelarrowexporter` and `otelarrowreceiver`
as alpha, adds them to the contribcol.

**Link to tracking Issue:** #26491

codeboten committed:
[chore] update solace receiver to use mdatagen (#33582)

This updates the code that was using OpenCensus to use mdatagen + otel

Fixes #33468

---------

Signed-off-by: Alex Boten <223565+codeboten@users.noreply.github.com>

dmitryax committed:
[chore] [receiver/mongodb] Fix test assertion (#33713)

The expected and actual errors are misplaced. It makes the test output
confusing

hardproblems committed:
Statsdreceiver optimizations (#33683)

**Description:**
Optimize statsdreceiver code to reduce heap usage
Also reduce failCnt increment to reduce memory footprint in cases of
tons of malformed statsd messages

**Link to tracking Issue:** <Issue number if applicable>

**Testing:**
- Tested internally and saw a 17% reduction in object allocation from
our workloads (with ~2.5k/s statsd metrics input) mostly from reduction
of strings.Split
Screenshot is from `go tool pprof -http:8081 -diff_base baseline.heap.gz
optimized.heap.gz`
![Screenshot 2024-06-20 at 12 45
17 PM](https://github.com/open-telemetry/opentelemetry-collector-contrib/assets/142453/e8065097-2533-4934-9b78-c0e828e075ac)

Note: I also saw lots of allocations from attribute.NewSet but idk the
best way to go about reducing that. (Screenshot below is from
unoptimized statsdreceiver)
![Screenshot 2024-06-20 at 12 44
32 PM](https://github.com/open-telemetry/opentelemetry-collector-contrib/assets/142453/293553c0-92c8-465b-872a-ddb7b68299a0)

I tried changing statsDMetricDescription's attrs field to
`map[string]any` in the hopes of using attributes.FromRaw in
buildGaugeMetric but it would require replacing statsDMetricDescription
as the type of the map key in instruments.gauges, etc.

martin-majlis-s1 committed:
Merge branch 'main' into datasetexporter-update-to-0.19.0
